airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala

```diff
@@ -774,7 +775,21 @@ object LogicalPlan {
     override def outputAttributes: Seq[Attribute] = mergeOutputAttributes
     protected def mergeOutputAttributes: Seq[Attribute] = {
       // Collect all input attributes
-      val outputAttributes: Seq[Seq[Attribute]] = children.map(_.outputAttributes.flatMap(_.inputColumns))
+      def collectInputAttributes(rels: Seq[Relation], result: Seq[Seq[Attribute]] = Nil): Seq[Seq[Attribute]] = {
```

Comment:
Collect input attributes recursively

Reply:
I don't think this code is recursively collecting input attributes from children nodes, rather it just skips collecting the outputAttributes from the root Set operation node.

A tricky example would be nested set operations like this:
```sql
select id from (select id, x from (select id, x, y from D) union all select id, x from B) union all select id from C?
```
In such a nested case, doesn't it collect attributes from a wrong query plan and report `id, x, y`?

Reply:
Recursively in my original comment may be wrong but can't we expect that children are already resolved when this method is called?

In this case, however, the output attributes become `((A.id, B.id), C.id)` and fail to merge because of lack of multiple input source columns handling. I fixed it and added test cases.